verification/certifyKey_test.go

```diff
@@ -3,61 +3,521 @@
+				//	vendorInfo 	[8] IMPLICIT OCTET STRING OPTIONAL,
+				var vendorInfo cryptobyte.String
+				if !diceTcbRaw.ReadASN1Element(&vendorInfo, cryptobyte_asn1.Tag(8).ContextSpecific()) {
+					msg := fmt.Sprintf("vendor info in dice tcb element %d is malformed", dindex)
+					return nil, errors.New(msg)
+				}
+				if !vendorInfo.ReadASN1(&vendorInfo, cryptobyte_asn1.Tag(8).ContextSpecific()) {
+					msg := fmt.Sprintf("vendor info in dice tcb element %d is malformed", dindex)
+					return nil, errors.New(msg)
+				}
+				d.VendorInfo = fmt.Sprintf("%x", vendorInfo)
+			} else if diceTcbRaw.PeekASN1Tag(cryptobyte_asn1.Tag(9).ContextSpecific()) {
+				// 	type 		[9] IMPLICIT OCTET STRING OPTIONAL,
+				var typeInfo cryptobyte.String
+				if !diceTcbRaw.ReadASN1Element(&typeInfo, cryptobyte_asn1.Tag(9).ContextSpecific()) {
+					msg := fmt.Sprintf("type info in dice tcb element %d is malformed", dindex)
+					return nil, errors.New(msg)
+				}
+				if !typeInfo.ReadASN1(&typeInfo, cryptobyte_asn1.Tag(9).ContextSpecific()) {
+					msg := fmt.Sprintf("type info in dice tcb element %d is malformed", dindex)
+					return nil, errors.New(msg)
+				}
+				d.Type = fmt.Sprintf("%x", typeInfo)
+			}
+
+			if len(diceTcbRaw) == 0 {
+				multiTcbInfo.DiceTcbInfos = append(multiTcbInfo.DiceTcbInfos, d)
+				break
+			}
+		}
+		if len(input) == 0 {
+			break
+		}
+		dindex++
+
+	}
+	return &multiTcbInfo, nil
+}
+
+// Check whether certificate extended key usage is as per spec
+// OID for ExtendedKeyUsage Extension: 2.5.29.37
+// The ExtendedKeyUsage extension SHOULD be marked as critical
+// If IsCA = true, the extension SHOULD contain tcg-dice-kp-eca
+// If IsCA = false, the extension SHOULD contain tcg-dice-kp-attestLoc
+func checkCertifyKeyExtendedKeyUsages(t *testing.T, c *x509.Certificate) {
+	t.Helper()
+
+	extKeyUsage := []asn1.ObjectIdentifier{}
+
+	for _, ext := range c.Extensions {
+		if ext.Id.Equal(OidExtensionExtKeyUsage) { // OID for ExtKeyUsage extension
+			// Extract the OID value from the extension
+			_, err := asn1.Unmarshal(ext.Value, &extKeyUsage)
+			if err != nil {
+				t.Errorf("[ERROR]: Failed to unmarshal the Extended Key Usage extension: %v", err)
+				continue
+			}
+
+			if !ext.Critical {
+				t.Errorf("[ERROR]: The Extended Key Usage extension IS NOT CRITICAL, MUST BE CRITICAL")
+			} else {
+				t.Logf("[LOG]: The Extended Key Usage extension is marked CRITICAL")
+			}
+			break
+		}
+	}
+
+	if len(extKeyUsage) == 0 {
+		t.Errorf("[ERROR]: The Extended Key Usage extension is empty")
+	}
+
+	// Iterate over the OIDs in the ExtKeyUsage extension
+	isExtendedKeyUsageValid := false
+	var expectedKeyUsage asn1.ObjectIdentifier
+	expectedKeyUsageName := ""
+	if c.IsCA {
+		expectedKeyUsage = OidExtensionTcgDiceKpEca
+		expectedKeyUsageName = "tcg-dice-kp-eca"
+	} else {
+		expectedKeyUsage = OidExtensionTcgDiceKpAttestLoc
+		expectedKeyUsageName = "tcg-dice-kp-attest-loc"
+	}
+
+	for _, oid := range extKeyUsage {
```

Comment:
can do `if slices.Contains(extKeyUsage, expectedKeyUsage)`

Reply:
Tried using slices, but it is throwing an error: 'encoding/asn1'.ObjectIdentifier does not satisfy comparability.'

Reply:
Ok, this is fine to leave then